src/create_update.c

```diff
@@ -212,6 +225,120 @@ static int check_build_env(void)
+	newm = g_hash_table_lookup(manifests->new_manifests, group);
+	add_component_hashes_to_manifest(newm, manifests->new_full);
+	apply_heuristics(oldm);
+	apply_heuristics(newm);
+	newm->prevversion = oldm->version;
+
+	/* add os-core as an included manifest */
+	if (!manifest_includes(oldm, "os-core")) {
+		oldm->includes = g_list_prepend(oldm->includes, manifests->old_core);
+	}
+	if (!manifest_includes(newm, "os-core")) {
+		newm->includes = g_list_prepend(newm->includes, manifests->new_core);
+	}
+
+	/* Step 5: Subtract the core files from the manifest */
+	G_LOCK(SUBTRACT);
+	subtract_manifests_frontend(oldm, oldm);
+	subtract_manifests_frontend(newm, newm);
+	G_UNLOCK(SUBTRACT);
+
+	/* Step 6: Compare manifest to the previous version... */
+	if (match_manifests(oldm, newm) == 0 && !changed_includes(oldm, newm)) {
+		LOG(NULL, "", "%s components have not changed, no new manifest", group);
+		printf("%s components have not changed, no new manifest\n", group);
+		/* Step 6a: if nothing changed, stay at the old version */
+		newm->version = oldm->version;
+	} else {
+		apply_heuristics(newm);
+#warning missing rename_detection here
+		old_deleted = remove_old_deleted_files(oldm, newm);
+		sort_manifest_by_version(newm);
+		type_change_detection(newm);
+		newfiles = prune_manifest(newm);
+		if (newfiles > 0 || old_deleted > 0 || changed_includes(oldm, newm)) {
+			LOG(NULL, "", "%s component has changes (%d new, %d deleted), writing out new manifest", group, newfiles, old_deleted);
+			printf("%s component has changes (%d new, %d deleted), writing out new manifest\n", group, newfiles, old_deleted);
+			if (write_manifest(newm) != 0) {
+				LOG(NULL, "", "%s component manifest write failed", group);
+				printf("%s component manifest write failed\n", group);
+				assert(0);
+			}
+			/* delta manifests will be created as a separate step. Since our pack creation
+			 * already creates a delta manifest, we can extend that functionality */
+			//create_manifest_deltas(newm, manifests->manifests_last_versions_list);
+		} else {
+			LOG(NULL, "", "%s component has not changed (after pruning), no new manifest", group);
+			printf("%s component has not changed (after pruning), no new manifest\n", group);
+			newm->version = oldm->version;
+		}
+	}
+	/* We have to lock to write to the MoM since there is only 1 copy of it */
+	G_LOCK(MOM);
+	nest_manifest(manifests->new_MoM, newm);
+	G_UNLOCK(MOM);
+}
+
+static void create_manifest_task(gpointer data, gpointer user_data)
+{
+	struct manifest_data *mlist = user_data;
+	char *group = data;
+	process_bundle(mlist, group);
+}
+
+static void submit_manifest_tasks(struct manifest_data *manifests)
+{
+	GThreadPool *threadpool;
+	int ret = 0;
+	int count = 0;
+	GError *err = NULL;
+	char *group;
+
+
+	printf("Manifest threadpool %ld threads\n", sysconf(_SC_NPROCESSORS_ONLN) * 3);
+	threadpool = g_thread_pool_new(create_manifest_task, manifests,
+					sysconf(_SC_NPROCESSORS_ONLN) * 3,
```

Comment:
minor issue, but the type of the third argument is "gint", but sysconf() returns a long int